tests/mpls/test_mpls.py

```diff
@@ -0,0 +1,594 @@
+LABEL_SWAP_ROUTES='label_swap_routes'
+LABEL_DEL_ROUTES='label_del_routes'
+
+@pytest.fixture(scope='module')
+def setup(duthost, tbinfo, ptfadapter):
+    """
+    setup fixture gathers all test required information from DUT facts and tbinfo
+    :param duthost: DUT host object
+    :param tbinfo: fixture provides information about testbed
+    :return: dictionary with all test required information
+    """
+    # gather ansible facts
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    int_facts = duthost.interface_facts()['ansible_facts']
+    host_facts = duthost.setup()['ansible_facts']
+
+    tor_ports_ids=[]
+    tor_ports={}
+    spine_ports_ids=[]
+    spine_ports={}
+    tor_addr={}
+    tor_peer_addr={}
+    spine_addr={}
+    spine_peer_addr={}
+    tor_mac={}
+    spine_mac={}
+ 
+    # get the list of TOR/SPINE ports
+    for dut_port, neigh in mg_facts['minigraph_neighbors'].items():
+        if not port_up(duthost, dut_port): 
+            continue
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        if 'T0' in neigh['name']:
+            tor_ports[port_id]=dut_port
+            tor_ports_ids.append(port_id)
+        elif 'T2' in neigh['name']:
+            spine_ports[port_id]=dut_port
+            spine_ports_ids.append(port_id)
+    
+    logger.info('tor_ports: {}'.format(tor_ports))
+    logger.info('spine_ports: {}'.format(spine_ports))    
+        
+    for blk in mg_facts['minigraph_interfaces']:
+        if ':' in blk['peer_addr']:
+            continue
+        dut_port=blk['attachto']
+        port_id = mg_facts['minigraph_port_indices'][dut_port] 
+        neigh=mg_facts['minigraph_neighbors'][dut_port]       
+        if 'T0' in neigh['name']:
+             tor_addr[port_id]=blk['addr']
+             tor_peer_addr[port_id]=blk['peer_addr']
+        elif 'T2' in neigh['name']:
+             spine_addr[port_id]=blk['addr']
+             spine_peer_addr[port_id]=blk['peer_addr']
+            
+    for pid, dut_port in tor_ports.items():
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        ansible_port='ansible_'+dut_port
+        tor_mac[port_id]=host_facts[ansible_port]['macaddress']
+    
+    for pid, dut_port in spine_ports.items():
+        port_id = mg_facts['minigraph_port_indices'][dut_port]
+        ansible_port='ansible_'+dut_port
+        spine_mac[port_id]=host_facts[ansible_port]['macaddress']
+
+    setup_information = {
+        'duthost': duthost,
+        'dut_tmp_dir': DUT_TMP_DIR,
+        'dst_ip_spine_blocked': '192.168.144.1',
+        'tor_ports': tor_ports,
+        'spine_ports': spine_ports,
+        'tor_addr': tor_addr,
+        'tor_peer_addr': tor_peer_addr,
+        'spine_addr': spine_addr,
+        'spine_peer_addr': spine_peer_addr,
+        'tor_ports_ids': tor_ports_ids,
+        'spine_ports_ids': spine_ports_ids,
+        'tor_mac': tor_mac,
+        'spine_mac': spine_mac,
```

Comment:
instead of returning all the ports how about returning just what the test needs i.e. one source port and destination port. 
```
setup_informaion = {
  .
  .
    test_params : { 
        'src-port-id': x,
        'dst-port-id: [a,b,c]    <--- in case of LAG port
        'src-IP': 
        'dst-IP':
    }
}
```
This way we can get rid of "get_src_portid", "get_dst_portid" APIs.

Reply:
Only for t1 topology. Not using portchannel/\.